server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -189,6 +190,9 @@ public IndexRequest(@Nullable ShardId shardId, StreamInput in) throws IOExceptio
                     : new ArrayList<>(possiblyImmutableExecutedPipelines);
             }
         }
+        if (in.getTransportVersion().onOrAfter(TransportVersions.SEMANTIC_TEXT_FIELD)) {
```

Comment:
IndexRequest has a pluginsPipeline field so it stays similar to the current request pipeline / default pipeline